libcontainer/process_linux.go

```diff
@@ -271,6 +296,30 @@ func (p *initProcess) createNetworkInterfaces() error {
 	return nil
 }
 
+func runCmd(command configs.Command, statePath string) error {
+	cmd := exec.Cmd{
+		Path: command.Path,
+		Args: command.Args[:],
+	}
+	cmd.Env = command.Env
+	cmd.Dir = command.Dir
+
+	stateFile, err := os.Open(statePath)
+	if err != nil {
+		return err
+	}
+	defer stateFile.Close()
+
+	cmd.Stdin = stateFile
```

Comment:
And I'd probably open `stateFile` first, and then setup `Stdin` in the `cmd := exec.Cmd{…}` line too.